api/services/ElasticSearchService.js

```diff
@@ -107,7 +133,49 @@ module.exports = {
     return RedisService.getJSONFromCache('percentiles', res, RedisService.DAILY, function() {
       return ElasticSearchService.lastMonthPercentiles();
     });
-  }
-
-
+  },
+  //download first 10000 results and proceed by processing and scrolling
+  lastMonthDownloadsBulk:function(days,callback){
+      hits = [];
+      var body = ElasticSearchService.queries.filters.lastMonthDownloads(days);
+      return es.search({
+      scroll:'5M',
+      index: 'stats',
+      body: body,
+      },function processAndGetMore(error,response){
+            //check the response
+        if (typeof response == "undefined" || typeof response.hits == "undefined") {    
+          console.log("you received an undefined response, response:"+response);
+          console.log("this was probably caused because there were no stats yet for this day");
+          console.log("or processing time took over 5 minutes (the scroll interval");
+        callback();
```

Comment:
In node, the convention when using callback is that the first argument is the error, end the second the result.
So when there was an error:
`callback(error)`,
when it's successful 
`callback(null, result)`.
Would be nice to send the an Error with a message when there is one, so here what you are logging can be the message, and you can use the basic `Error` object:
https://nodejs.org/api/errors.html#errors_class_error